packages/js/product-editor/src/components/catalog-visibility/catalog-visibility.tsx

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import { createElement, Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { recordEvent } from '@woocommerce/tracks';
+import { CheckboxControl } from '@wordpress/components';
+import { ProductCatalogVisibility } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import { CatalogVisibilityProps } from './types';
+import { TRACKS_SOURCE } from '../../constants';
+
+export function CatalogVisibility( {
+	catalogVisibility,
+	onCheckboxChange,
+}: CatalogVisibilityProps ) {
+	function handleVisibilityChange(
+		selected: boolean,
+		visibility: ProductCatalogVisibility
+	) {
+		if ( selected ) {
+			if ( catalogVisibility === 'visible' ) {
+				onCheckboxChange( visibility );
+				recordEvent( 'product_catalog_visibility', {
+					source: TRACKS_SOURCE,
+					visibility: catalogVisibility,
+				} );
+				return;
+			}
+			onCheckboxChange( 'hidden' );
+		} else {
+			if ( catalogVisibility === 'hidden' ) {
+				if ( visibility === 'catalog' ) {
+					onCheckboxChange( 'search' );
+					recordEvent( 'product_catalog_visibility', {
+						source: TRACKS_SOURCE,
+						visibility: catalogVisibility,
+					} );
+					return;
+				}
+				if ( visibility === 'search' ) {
+					onCheckboxChange( 'catalog' );
+					recordEvent( 'product_catalog_visibility', {
+						source: TRACKS_SOURCE,
+						visibility: catalogVisibility,
+					} );
+					return;
+				}
+				return;
+			}
+			onCheckboxChange( 'visible' );
+			recordEvent( 'product_catalog_visibility', {
+				source: TRACKS_SOURCE,
+				visibility: catalogVisibility,
+			} );
+		}
+	}
+
+	return (
+		<>
+			<CheckboxControl
+				label={ __( 'Hide in product catalog', 'woocommerce' ) }
+				checked={
+					catalogVisibility === 'search' ||
+					catalogVisibility === 'hidden'
+				}
+				onChange={ ( selected ) =>
+					handleVisibilityChange( selected, 'search' )
+				}
+			/>
+			<CheckboxControl
+				label={ __( 'Hide from search results', 'woocommerce' ) }
+				checked={
+					catalogVisibility === 'catalog' ||
+					catalogVisibility === 'hidden'
+				}
+				onChange={ ( selected ) =>
+					handleVisibilityChange( selected, 'catalog' )
+				}
```

Comment:
How come you changed this to two checkboxes? couldn't you keep this as a singular checkbox the same as the block worked as a singular checkbox?

Reply:
And pass the `visibility` prop in.

Reply:
Now I get the idea! I changed that in the commit https://github.com/woocommerce/woocommerce/pull/44434/commits/69d9ee5b51bf223c164c984805db03f519aea955